pkg/controller/workflow/executor.go

```diff
@@ -405,7 +405,10 @@ func (c *workflowExecutor) HandleFlyteWorkflow(ctx context.Context, w *v1alpha1.
 			return err
 		}
 		if err := c.TransitionToPhase(ctx, w.ExecutionID.WorkflowExecutionIdentifier, wStatus, newStatus); err != nil {
-			return err
+			// Ignore ExecutionNotFound error to allow graceful failure
+			if err != nil && !eventsErr.IsNotFound(err) {
+				return err
+			}
```

Comment:
There is one more failure scenario we need to handle here. There is something called FailureNode (which is sort of the finally{} block in a try/catch block that's only called on failure (Mainly to cleanup up resources... etc.).

If there is a FailureNode defined, it'll transition HandlingFailureNode which will then try to start the node (transition that to Queued which will fail) then it'll get stuck at that phase...

I think if you add this block to the next `switch case v1alpha1.WorkflowHandlingFailureNode` [below ](https://github.com/flyteorg/flytepropeller/blob/6a14e7fbffe89786fb1d8cde22715f93c2f3aff5/pkg/controller/workflow/executor.go#L418) that would be enough.